Adding SPARQL tests to the family reference data pipeline.

robons opened the pull request:
I've slightly refactored the family transform pipeline using shared pipeline [custom steps](https://www.jenkins.io/doc/book/pipeline/shared-libraries/#defining-custom-steps). 

I've not been able to test this locally given we mock the drafter. Let's see whether it works shall we? 

https://github.com/GSS-Cogs/pmd-jenkins-library/issues/44

ajtucker reviewed (changes requested):
Good stuff, thanks.

I've tried to move from putting new commands in `vars` to adding them as methods to classes or singletons in the main code, as otherwise they share a global namespace. It's not super important.

We could do with parameterising the SPARQL tests function such that we can choose which tests/directories to run. The reference data shouldn't be doing Cube things and so shouldn't need to run the `qb` tests, and I'd also like to be able to potentially have some ICs local to the family or dataset, though that's for later.

robons commented:
> Good stuff, thanks.
> 
> I've tried to move from putting new commands in `vars` to adding them as methods to classes or singletons in the main code, as otherwise they share a global namespace. It's not super important.
> 
I've moved the small ones into the `util` class but more or less left the larger SPARQL tests one in its own (now namespaced) class.

> We could do with parameterising the SPARQL tests function such that we can choose which tests/directories to run. The reference data shouldn't be doing Cube things and so shouldn't need to run the `qb` tests, and I'd also like to be able to potentially have some ICs local to the family or dataset, though that's for later.

I've updated the code so that the SPARQL test step takes an optional list of which tests to run. If this isn't provided, it runs all three. If we need to add dataset specific tests, we can add another Enum value, decide on where they are going to be bindmounted in the docker container and point the Enum value to that directory.

ajtucker reviewed (approved):
Looks good.

As always, there's some close coupling between the docker containers and these scripts in that the library needs to know the layout of the tests in gdp-sparql-tests, but we just need to remember for now!
